pkg/cluster/spec/instance.go

```diff
@@ -212,6 +214,13 @@ func (i *instance) InstanceName() string {
 
 // ServiceName implements Instance interface
 func (i *instance) ServiceName() string {
+	switch i.ComponentName() {
+	case ComponentSpark, ComponentTiSpark:
+		if i.port > 0 {
+			return fmt.Sprintf("%s-%d.service", i.Role(), i.port)
+		}
+		return fmt.Sprintf("%s.service", i.Role())
```

Comment:
I think we should specify a default port in the Unmarshal stage. So `i.port > 0` should always be true.

Reply:
It does, if we defined a default value for port in the spec tag.